merkletree/proof.go

```diff
@@ -80,38 +80,51 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-func (ap *AuthenticationPath) verifyBinding(key, value []byte) bool {
+// VerifyBinding verifies both the value and the commitment
+// of the proof node of the authentication path.
+//
+// This should be called only when ap is a proof of inclusion.
+func (ap *AuthenticationPath) VerifyBinding(key, value []byte) bool {
```

Comment:
Sorry, one more question: Should the server be setting `proofType = undeterminedProof` explicitly in `merkltree.Get`? It seems like a server could set `proofType` to whatever it wanted and `ap.ProofType()` wouldn't validate that setting except for in the case of `undeterminedProof`.